File: app/src/main/java/com/mytlogos/enterprise/background/api/model/AddClientExternalUser.java
package com.mytlogos.enterprise.background.api.model;

public class AddClientExternalUser extends ClientExternalUser {
    private String pwd;

    public AddClientExternalUser(String uuid, String identifier, int type, ClientExternalMediaList[] lists, String pwd) {
        super("", uuid, identifier, type, lists);
        this.pwd = pwd;
    }

    public String getPwd() {
        return pwd;
    }

    @Override
    public String toString() {
        return "AddClientExternalUser{" +
                "pwd='" + pwd + '\'' +
                '}';
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        AddClientExternalUser that = (AddClientExternalUser) o;

        return getPwd() != null ? getPwd().equals(that.getPwd()) : that.getPwd() == null;
    }

    @Override
    public int hashCode() {
        return getPwd() != null ? getPwd().hashCode() : 0;
    }
}

File: app/src/main/java/com/mytlogos/enterprise/background/api/model/AddClientExternalUser.kt
package com.mytlogos.enterprise.background.api.model

/**
 * TODO: check Parameter for POST externalUser.
 */
class AddClientExternalUser(uuid: String, identifier: String, type: Int, lists: Array<ClientExternalMediaList>, private val pwd: String) : ClientExternalUser("", uuid, identifier, type, lists) {
    fun getPwd(): String? {
        return pwd
    }

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false
        if (!super.equals(o)) return false
        val that = o as AddClientExternalUser
        return if (getPwd() != null) getPwd() == that.getPwd() else that.getPwd() == null
    }

    override fun hashCode(): Int {
        var result = super.hashCode()
        result = 31 * result + if (getPwd() != null) getPwd().hashCode() else 0
        return result
    }
}